libs/domain/picking/src/services/adapter/picking-list-default.adapter.ts

```diff
@@ -41,12 +42,25 @@ export class PickingListDefaultAdapter implements PickingListAdapter {
     return this.pickingHttpService
       .patch<StartPickingListResponse>(`/picking-lists/${pickingList.id}`, body)
       .pipe(
-        map(({ data: updatedPickingListData }) => ({
-          ...pickingList,
-          status: updatedPickingListData.status as PickingListStatus,
-          createdAt: new Date(updatedPickingListData.createdAt),
-          updatedAt: new Date(updatedPickingListData.updatedAt),
-        }))
+        map((response) => {
+          if (response.errors) {
+            for (let i = 0; i < response.errors.length; i++) {
+              const error = response.errors[i];
+              const pickingListError = new Error(
+                error.message
+              ) as PickingListError;
+              pickingListError.status = error.status;
+              pickingListError.code = error.code;
+              throw pickingListError;
```

Comment:
Because you are throwing an error immediately you do not need a for loop, just grab the first element directly.